SimPEG/electromagnetics/static/utils/StaticUtils.py

```diff
@@ -1686,56 +1687,15 @@ def gettopoCC(mesh, actind, option="top"):
             return mesh1D, topoCC
 
     elif mesh._meshType == "TREE":
-        if mesh.dim == 3:
-            core_inds = np.isin(
-                mesh.h_gridded,
-                np.r_[mesh.hx.min(), mesh.hy.min(), mesh.hz.min()]
-            ).all(axis=1)
-
-            act_core_inds = actind[core_inds]
-
-            uniqXY = uniqueRows(mesh.gridCC[core_inds, :2])
-            npts = uniqXY[0].shape[0]
-            ZC = mesh.gridCC[core_inds, 2]
-            topoCC = np.zeros(npts)
-            if option == "top":
-                # TODO: this assume same hz, need to be modified
-                dz = mesh.hz.min() * 0.5
-            elif option == "center":
-                dz = 0.
-            for i in range(npts):
-                inds = uniqXY[2] == i
-                actind_z = act_core_inds[inds]
-                if actind_z.sum() > 0.:
-                    topoCC[i] = (ZC[inds][actind_z]).max() + dz
-                else:
-                    topoCC[i] = (ZC[inds]).max() + dz
-            return uniqXY[0], topoCC
-        else:
-            core_inds = np.isin(
-                mesh.h_gridded,
-                np.r_[mesh.hx.min(), mesh.hy.min()]
-            ).all(axis=1)
-
-            act_core_inds = actind[core_inds]
-
-            uniqX = np.unique(mesh.gridCC[core_inds, 0], return_index=True, return_inverse=True)
-            npts = uniqX[0].shape[0]
-            ZC = mesh.gridCC[core_inds, 1]
-            topoCC = np.zeros(npts)
-            if option == "top":
-                # TODO: this assume same hz, need to be modified
-                dy = mesh.hy.min() * 0.5
-            elif option == "center":
-                dy = 0.
-            for i in range(npts):
-                inds = uniqX[2] == i
-                actind_z = act_core_inds[inds]
-                if actind_z.sum() > 0.:
-                    topoCC[i] = (ZC[inds][actind_z]).max() + dy
-                else:
-                    topoCC[i] = (ZC[inds]).max() + dy
-            return uniqX[0], topoCC
+
+        inds = mesh.get_boundary_cells(actind, direction='zu')[0]
+
+        if option == "top":
+            dz = mesh.h_gridded[inds, 2] * 0.5
```

Comment:
Will this fail in 2D? Should probably replace with `dz = mesh.h_gridded[inds, -1] * 0.5`